python/examples/matmul/test_schedule_dialect.py

```diff
@@ -0,0 +1,69 @@
+# RUN: %PYTHON %s 2>&1 | FileCheck %s
+
+# This file contains small benchmarks with reasonably-sized problem/tiling sizes
+# and codegen options.
+
+import mlir.iree_sandbox as sandbox
+import mlir.ir as ir
+import mlir.dialects.pdl as pdl
+import mlir.dialects.linalg_transform as transform
+
+from ..core.experts import *
+from ..core.harness import *
+from ..core.transforms import *
+
+from ..contraction.definitions import EinsumProblem
+
+################################################################################
+### Problem instantiations.
+################################################################################
+
+keys = ['m', 'n', 'k']
+
+# CHECK-NOT: FAILURE
+
+
+def add_tile_schedule(module):
+  with InsertionPoint(module.body):
+    pdl_pattern_name = 'pdl_pattern_name'
+    pdl_pattern = pdl.PatternOp(\
+      1, #=benefit 
+      pdl_pattern_name#=sym_name
+    )
+    with ir.InsertionPoint(pdl_pattern.body):
+      args = pdl.OperandsOp()
+      types = pdl.TypesOp()
+      pdl_op = pdl.OperationOp('linalg.generic', args=[args], types=[types])
+      pdl.RewriteOp(pdl_op, 'linalg_transform.apply')
+
+    sequence = transform.SequenceOp()
+    with ir.InsertionPoint(sequence.body.blocks[0]):
+      transform.TileOp(pdl_pattern_name, sizes=[12, 32, 1], pad=False)
```

Comment:
Side note: If you want the same behavior as codegen strategy wrt padding, you'd have to write two independent `TileOp`s -- one to tile, another to pad the result of the previous tiling. Otherwise, the enabler pass won't run between them which _may_ produce different padded shapes.

Reply:
ack